src/redux/Courier/taskActions.js

```diff
@@ -168,19 +167,13 @@ function uploadTasksImages(tasks, state) {
     return Promise.resolve()
   }
 
-  const promises = files.map(file => uploadTaskImage(state.app.httpClient, file))
+  const promises = files.map(file => uploadTaskImage(state.app.httpClient, file, {
+    headers: {
+      'X-Attach-To': tasks.map(task => task['@id']).join(';'),
+    },
+  }))
 
   return Promise.all(promises)
-    .then(values => {
-      if (values.length === 0) {
-        return tasks
-      }
-      // Associate images with task
-      return state.app.httpClient.put('/api/tasks/images', {
-        images: values.map(image => image['@id']),
-        tasks: tasks.map(task => task['@id']),
-      })
-    })
```

Comment:
Servers must be updated before deploying this change.
The server change is retro-compatible.
And i updated the server part since then -> https://github.com/coopcycle/coopcycle-web/commit/5328c7fb23d3512639193ad43bd04506adc4fcfd